README.md

```diff
@@ -0,0 +1,705 @@
+#### Step: Build images
+
+The [docker/build-push-action@v5](https://github.com/docker/build-push-action) is used to build images. This step builds the images from corresponded [`Containerfile`](https://github.com/AlmaLinux/container-images/tree/main/Containerfiles), for specified `env.platforms` and using tags from the previous step. After the successful building, the images are loaded into docker, and not yet pushed because they need to be tested. AlmaLinux 8 minimal images buildx looks like:
+```sh
+/usr/bin/docker buildx build --file ./Containerfile.minimal ... \
+ --platform linux/amd64,linux/ppc64le,linux/s390x,linux/arm64 \
+ --provenance false ... \
+ --tag docker.io/***/8-minimal:latest --tag docker.io/***/8-minimal:8 --tag docker.io/***/8-minimal:8.9 --tag docker.io/***/8-minimal:8.9-20240319 --tag quay.io/***/8-minimal:latest --tag quay.io/***/8-minimal:8 --tag quay.io/***/8-minimal:8.9 --tag quay.io/***/8-minimal:8.9-20240319 \
+ --load \
+ --metadata-file /home/runner/work/_temp/docker-actions-toolkit-*/metadata-file \
+ https://github.com/***/container-images.git#270a6d3fd433cfa6c3e1fff5896a92d1ae2896be:Containerfiles/8
+```
+`provenance: false` is to disable the [Provenance attestations](https://docs.docker.com/build/attestations/slsa-provenance/) as Quay.io registry doesn't support such kind of images data.
+
+#### Step: Test images
+
+As every image is loaded into docker, so it can be tested separately for each type and platform. Docker run images by digest:
+```sh
+docker run --platform=${platform} ${{ steps.build-images.outputs.digest }}
+```
+
+#### Step: Push images to Client Library
+
+The [docker/build-push-action@v5](https://github.com/docker/build-push-action) is used. This step pushes built images into *Client Library*. The options are the same as for **Build images** step.
+
+
+#### Step: Change date stamp in Containerfile (default and minimal only)
+
+> The step is skipped if image type is not 'default' or 'minimal'
+
+The step changes (*# Tags* with date stamp) corresponded [`Containerfiles/*/Containerfile.*`](https://github.com/AlmaLinux/container-images/tree/docker-library), which Docker will use to build images for the *Official Library*. An example is for AlmaLinux 8 minimal `Containerfiles/8/Containerfile.minimal` file:
+```docker
+# Tags: 8-minimal, 8.9-minimal, 8.9-minimal-20240319
+
+FROM quay.io/almalinuxorg/8-minimal:8.9-20240319
+```
+The change indicates that new `default` and/or `minimal` container image was pushed to *Client Library* and the same should be requested to build by Docker. The change will later be committed into `docker-file` branch.
+
+#### Step: Upload changed *Containerfiles/*/Containerfile.**
+
+> The step is skipped if image type is not 'default' or 'minimal'
+
+The step uses [actions/upload-artifact@v4](https://github.com/actions/upload-artifact) to store as artifact changed on previous step Containerfile. The artifact is named against `image_type`, like `containerfiles-image_types`.
```

Comment:
The step uses [actions/upload-artifact@v4](https://github.com/actions/upload-artifact) to store the artifact changed in the previous Containerfile step. The artifact is named against `image_type`, like `containerfiles-image_types`.

Reply:
Done.